### ISSUE TITLE
[🐛BUG] 模型KSR的实现与原论文不一致。

### ISSUE THREAD

**Issue by WYH-han** [RUCAIBox/RecBole#1630]
**描述这个 bug**
1. 本KSR项目中，memory_read函数（即对应原论文的公式(6)）中计算attentions，应采用“gru_output”（公式中的"~h_ut"）而非“user_memory”(公式中的"v_ua");
2. 原论文中item_embedding采用BRP预训练得到，章节5.1的Parameter Setting中提到：item_embedding的维度与KG_embedding不一致。而本KSR项目实现中将二者统一用“embedding_size”参数设置了。

**预期**
针对问题1：ksr.py中memory_read函数中修改为：
`attentions = nn.functional.softmax(
            self.gamma * torch.mul(gru_output, attrs).sum(-1).float(), -1
        )  # [B R]`
这一步计算之前需要对gru_output的维度变更。目前不太清楚该如何将包含sequence长度与包含relation数量的两个矩阵进行怎样的处理？

针对问题2：增加参数item_embedding_size用于设置项目嵌入的维度，embedding_size仅针对KG嵌入的维度设置。

**实验环境（请补全下列信息）：**
 - 操作系统: macOS 
- RecBole 版本 [如 1.1.1]
 - Python 版本 [如 3.7.13]
- PyTorch 版本 [如 1.12.1]

**Comment by NietzscheMYB** [comment]
您好！我想请问一下在KSR模型的初始化中，为什么self.entity_embedding的定义为`self.entity_embedding = nn.Embedding(
            self.n_items, self.embedding_size, padding_idx=0)`，不应该是`self.entity_embedding = nn.Embedding(self.n_entities, self.embedding_size, padding_idx=0)`

**Comment by zhengbw0324** [comment]
@WYH-han 您好！
感谢您的关注！参考您的建议和原论文模型描述，我们在[#1647](https://github.com/RUCAIBox/RecBole/pull/1647)对KSR模型进行了完善。

**Comment by zhengbw0324** [comment]
@NietzscheMYB 您好！
因为并不是所有知识图谱实体都与item对应，因此我们仅从预训练的entity embedding中载入了对推荐有意义的部分。您可以参考下方代码：
https://github.com/RUCAIBox/RecBole/blob/029c318dc233c299e0d738f0e749e90ebe2fc9d4/recbole/model/sequential_recommender/ksr.py#L83-L85

**Comment by NietzscheMYB** [comment]
> @NietzscheMYB 您好！ 因为并不是所有知识图谱实体都与item对应，因此我们仅从预训练的entity embedding中载入了对推荐有意义的部分。您可以参考下方代码：
> 
> https://github.com/RUCAIBox/RecBole/blob/029c318dc233c299e0d738f0e749e90ebe2fc9d4/recbole/model/sequential_recommender/ksr.py#L83-L85

哦哦了解了，感谢大佬！

**Comment by ljf012** [comment]
> @NietzscheMYB 您好！ 因为并不是所有知识图谱实体都与item对应，因此我们仅从预训练的entity embedding中载入了对推荐有意义的部分。您可以参考下方代码：
> 
> https://github.com/RUCAIBox/RecBole/blob/029c318dc233c299e0d738f0e749e90ebe2fc9d4/recbole/model/sequential_recommender/ksr.py#L83-L85

您好，我想知道KSR模型所需的该预训练的实体embedding可以从哪里获取？
我在[https://drive.google.com/drive/folders/1B5Bacbli0G9qW7Tm0H039eKww8rwKPUY](KGDatasets)中下载了KG数据并按照[https://github.com/RUCAIBox/RecSysDatasets/blob/master/conversion_tools/usage/MovieLens-KG.md](conversion_tools)处理了MovieLen和Amazon的数据，但并没有找到能训练其embedding的指引？
